### PR TITLE
Modify huggingface_model_json.ipynb

### PR DESCRIPTION
**Pull request by riboyuan99**
Add a function to calculate number of words(tokens) in a string

Modification on Benchmark: output tokens/second for different batch sizes(1,8,64)

Saved outputs as .pickle files for future re-use.

### CODE REVIEW

**Comment by goldmermaid** [comment]
can you remove the `.pickle` files?

**Comment by goldmermaid** [comment]
actually this PR needs more major refactoring... i will close it and you can post a new PR

**Comment by goldmermaid** [comment]
The experiment results are valuable though, just paste here


> We benchmarked to see the optimal batch_size for the TransformQAHuggingFaceJsonFormatConfig flow. The answer is "It depends on your data token length, your GPU memory, your LLM size, etc." In the following experiment, we use an AWS g5.xlarge instance that has a GPU with 24G memory and a quantized LLM (2G). We still use the above raw data strings raw_context_input.


- batch_size = 1
  100%|██████████| 1000/1000 [2:01:13<00:00, 7.27s/it]
  output tokens = 157250
  input tokens = 57000
  prompt tokens = 32
  29.5 tokens/second

- batch_size = 8
  100%|██████████| 125/125 [1:08:59<00:00, 13.20s/it]
  output tokens = 178250
  input tokens = 57000
  prompt_tokens = 32
  56.8 tokens/second

- batch_size = 64
  100%|██████████| 16/16 [10:27<00:00, 39.24s/it]
  output tokens = 173010
  input tokens = 57000
  prompt_tokens = 32
  366.9 tokens/second

File: example/transform/huggingface_model_json.ipynb [hunk 1]
```diff
@@ -190,42 +164,68 @@
...
+      "Prompt:  \n",
+      "        Generate one question and its corresponding answer based on the last context in the last\n",
+      "        example. Follow the format of the examples below to include context, question, and answer in the response.\n",
+      "        \n"
      ]
     }
    ],
    "source": [
-    "print(config)"
+    "from pprint import pprint\n",
```
Comment (SayaZhang) [inline example/transform/huggingface_model_json.ipynb:223]: Repeated import `pprint` here

File: example/transform/huggingface_model_json.ipynb [hunk 2]
```diff
@@ -141,38 +121,32 @@
...
+   "execution_count": 6,
    "metadata": {},
    "outputs": [
     {
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "sample size of processed input data:  4\n"
+      "sample size of processed input data:  1000\n",
+      "Example uniflow context data:\n",
```
Comment (SayaZhang) [inline example/transform/huggingface_model_json.ipynb:132]: Nit: too much output here, and the output content is a bit difficult to distinguish.

**Review by goldmermaid — APPROVED** [review]
LGTM

**Pull request merged**